### PR TITLE
Add one more method to CalibrationStore and proper unit tests for the refactored use-cases

### PR DESCRIPTION
**Pull request by gdetrez**
- Add a method to `CalibrationStore` to get all the measurements for a CCC.
- Add unit tests to the refactored use cases (`add_ccc`, `activate`ccc`, `delete_ccc` and `construct_polynomial`) that mock the calibration store.

### CODE REVIEW

File: scanomatic/data_processing/calibration.py [hunk 1]
```diff
@@ -495,21 +495,15 @@ def validate_polynomial(slope, p_value, stderr):
     return CalibrationValidation.OK
 
 
-def _collect_all_included_data(ccc):
+def _collect_all_included_data(store, calibrationid):
```
Comment (local-minimum) [inline scanomatic/data_processing/calibration.py]: why not stick with `identifier` as in the other functions or at least just have one name for the same identifier in the module.

File: scanomatic/data_processing/calibration.py [hunk 2]
```diff
@@ -604,8 +598,7 @@ def calculate_polynomial(data_store, degree=5):
 @_validate_ccc_edit_request
 def construct_polynomial(store, identifier, power):
 
-    ccc = store.get_calibration_by_id(identifier)
-    data_store = _collect_all_included_data(ccc)
+    data_store = _collect_all_included_data(store, identifier)
```
Comment (local-minimum) [inline scanomatic/data_processing/calibration.py]: I'm wondering if it becomes a bit more confusing now that there's a `store` and a `data_store` (which is more a collection of some data) ...

File: tests/unit/data_processing/test_calibration.py [hunk 3]
```diff
@@ -932,3 +939,239 @@ def test_doesnt_scramble_data_while_sorting(self, store, edit_ccc):
...
+        ccc[CellCountCalibration.polynomial] = (
+            ccc_data.get_polynomal_entry(len(polynomial) - 1, polynomial)
+        )
+    ccc[CellCountCalibration.edit_access_token] = access_token
+    if active:
+        ccc[CellCountCalibration.status] = CalibrationEntryStatus.Active
+    else:
+        ccc[CellCountCalibration.status] = (
+            CalibrationEntryStatus.UnderConstruction
+            )
```
Comment (local-minimum) [inline tests/unit/data_processing/test_calibration.py]: indentation errror?

File: tests/unit/data_processing/test_calibration.py [hunk 4]
```diff
@@ -932,3 +939,239 @@ def test_doesnt_scramble_data_while_sorting(self, store, edit_ccc):
...
+        ccc = make_calibration(identifier=None)
+        assert calibration.add_ccc(store, ccc) is False
+        store.add_calibration.assert_not_called()
+
+    def test_duplicate_id(self):
+        store = MagicMock(calibration.CalibrationStore)
+        store.has_calibration_with_id.return_value = True
+        ccc = make_calibration()
+        calibration.add_ccc(store, ccc)
+        assert store.add_calibration.called_with(ccc)
```
Comment (local-minimum) [inline tests/unit/data_processing/test_calibration.py]: What is it that happens here? The newer overwrites the older?

File: tests/unit/data_processing/test_calibration.py (hunk 3 above)
Comment (gdetrez) [reply tests/unit/data_processing/test_calibration.py]: No. Might be a code style but I'm not sure we have a style guide documented anywhere for scan-o-matic...

File: tests/unit/data_processing/test_calibration.py (hunk 4 above)
Comment (gdetrez) [reply tests/unit/data_processing/test_calibration.py]: No, just a bad assertion. Fixed.

**Pull request merged**